app/models/ability.rb

```diff
@@ -182,7 +182,7 @@ def message_abilities
         message.offer.created_by_id == @user_id && !message.is_private
       end
     end
-    can [:mark_read], Message, id: @user.subscriptions.pluck(:message_id)
+    can [:mark_read, :mark_all_read], Message, id: @user.subscriptions.pluck(:message_id)
```

Comment:
`mark_all_read` is a route collection action not a member e.g. `/messages/mark_all_read` rather than `/messages/1/mark_all_read` so in this case the ability for a user to mark all their messages as read can be given to every logged in user. You are scoping the messages in the controller query to just `User.current_user` so that's ok.

```suggestion
    can [:mark_read], Message, id: @user.subscriptions.pluck(:message_id)
    can [:mark_all_read]
```